Create manifest.json

msjsc001 opened the pull request:
A simple plugin to count words (Chinese & English) on the current Logseq page.

# Submit a new Plugin to Marketplace

**Plugin Github repo URL:** https://github.com/msjsc001/logseq-word-count-plugin

## Github releases checklist

- [x] a legal [package.json](https://gist.github.com/xyhp915/bb9f67f5b430ac0da2629d586a3e4d69#explain-packagejson) file.
- [x] a [valid CI workflow](https://github.com/xyhp915/logseq-journals-calendar/blob/main/.github/workflows/publish.yml#L10) build action for Github releases. (theme plugin for [this](https://github.com/Sansui233/logseq-bonofix-theme/blob/master/.github/workflows/publish.yml)).
- [x] a release which includes a release zip pkg from a successful build.
- [x] a clear README file, ideally with an image or gif showcase. (For more friendly to users, it is recommended to have English version description).
- [x] a license in the LICENSE file.

xyhp915 commented:
Hi @msjsc001 I found this plugin is using `index.js` as the entry point, but it has bug for the final Desktop version(0.10.9). Could you temporarily adjust the entry to `index.html`? https://github.com/msjsc001/logseq-word-count-plugin/blob/main/package.json#L4

msjsc001 commented:
@xyhp915 
![image](https://github.com/user-attachments/assets/27ecc9cf-1848-44e7-baa6-50f39f87d543)
Hi @xyhp915,

Thank you for reviewing the plugin and for the suggestion to change the entry point to index.html for compatibility with Logseq >= 0.10.9.

I've updated the plugin according to your suggestion. The package.json now specifies "main": "index.html", and I've adjusted the Vite build configuration accordingly (switching from library mode to application mode).

However, after making these changes, I'm consistently encountering loading issues when testing with Logseq 0.10.9, even in a completely clean virtual machine environment (Windows, Logseq 0.10.9, no other plugins, cleared cache).

The errors shown in the developer console (screenshots attached previously/below) include:



Failed to load resource: net::ERR_FILE_NOT_FOUND file:///C:/index.ts:1 (This path seems incorrect and unrelated to the plugin's actual location)

Loadhandshake Timeout Error: handshake Timeout

DOMException: Blocked a frame with origin "app://logseq.com" from accessing a cross-origin frame.

Uncaught TypeError: Cannot read properties of null (reading '$cljs$core$IFn$_invoke$arity$1$')

The warning 'This plugin #logseq-word-count-plugin takes too long to load...'

I've taken some steps.:


Different methods for updating the UI (using the Settings API instead of direct DOM manipulation).

Adjusting Vite build options (target environment (es2020), disabling minification, disabling sourcemaps, explicitly setting @logseq/libs as external).

Simplifying the plugin code to a bare minimum logseq.ready() call with a console.log.

Despite these efforts, the same errors persist.

The original version of the plugin, which used index.js as the entry point (built with Vite in library mode, iife format), loaded and functioned correctly without these errors in the same Logseq version.

Given that these errors persist in a clean environment specifically after switching to the index.html entry point, and especially considering the strange C:/index.ts file path error, I suspect there might be an underlying compatibility issue between Logseq 0.10.9 and how it loads plugins built with Vite using an HTML entry point.

Could you perhaps offer further guidance on how to resolve this, or would reverting to the index.js entry point (which worked previously) be acceptable for the marketplace?

Thank you again for your time and help!

msjsc001 commented:
> Hi @msjsc001 I found this plugin is using `index.js` as the entry point, but it has bug for the final Desktop version(0.10.9). Could you temporarily adjust the entry to `index.html`? https://github.com/msjsc001/logseq-word-count-plugin/blob/main/package.json#L4

Hello, I've spent a significant amount of time testing numerous different methods and referencing over ten existing community plugins. As long as an HTML entry point is used, two problems arise: 1. A prompt indicates slow loading, after which the plugin becomes unusable (I also noticed that many plugins already available in the community now encounter this same issue). 2. If the loading issue is resolved, it leads to the problem where the JS completely fails to execute. I suspect there might be compatibility issues with the official platform. Below is the detailed content of my tests for your reference and guidance:

# Investigation of JS Code Not Executing in Logseq Plugin HTML Entry Mode (logseq-word-count-plugin on v0.10.9)

**Date:** 2025-04-16

**Core Issue:** When attempting to adapt `logseq-word-count-plugin` for Logseq >= 0.10.9 (which requires an `index.html` entry point), despite successful builds using various tools (Vite, Rollup, Parcel) and Logseq recognizing the plugin (icon visible), the plugin's JavaScript code consistently **does not execute at all** under the standard loading method (no functionality, no console logs), even when extremely simplified. Meanwhile, other plugins using similar tech stacks work correctly in the same environment. The problem is strongly suspected to be a subtle interaction Bug or conflict between this specific project's build output and Logseq 0.10.9 under certain conditions.

**Goal:** (Original Goal) Modify the plugin entry point from `index.js` to `index.html` according to Logseq official marketplace review requirements (PR #639, by xyhp915).

**Final Status:** Unable to resolve the JS non-execution issue within the current project. Further debugging has been abandoned, and this document records the investigation process.

------

## Detailed Investigation Process and Timeline

1.  **Initial State:** Plugin used `index.js` entry, functioned correctly, but needed adaptation for marketplace requirements.
2.  Switched to Vite + HTML entry (Initial attempt):
    *   Modified `package.json` (added root `main: dist/index.html`, removed `logseq.main`).
    *   Modified `vite.config.ts` (removed `build.lib`).
    *   Modified `index.html` (removed manual script import, added mount point and `type="module"` script import).
    *   **Issue:** When testing via 'Load unpacked plugin' in Logseq 0.10.9 (VM), encountered `Loadhandshake Timeout Error` and a strange `ERR_FILE_NOT_FOUND file:///C:/index.ts:1` error. Standard loading method test also failed.
3.  Switched to Rollup + HTML entry:
    *   Uninstalled Vite, installed Rollup and related plugins.
    *   Created `rollup.config.js`, configured output to `dist/index.html`.
    *   Modified `package.json` (build script changed to `rollup -c`, tried various entry point configurations).
    *   Fixed errors during Rollup build (JSON import issues, TS compilation exclusions).
    *   **Issue:**
        *   Got `Illegal Logseq plugin package` error when using 'Load unpacked plugin'.
        *   Got `Loadhandshake Timeout Error` when testing the `.zip` package using the standard loading method (Method C).
4.  Researched other plugin examples:
    *   Downloaded and analyzed `logseq-plugin-split-block`, `logseq-plugin-bullet-threading`, `logseq-pomodoro-plugin`, `logseq-plugin-tabs`, `logseq-pomodoro-timer` (official example).
    *   Findings:
        *   **Conflicting Entry Point Config:** Third-party plugins often omit `logseq.main`; official examples define `logseq.main`. Conclusion: No single correct configuration.
        *   **Build Tools:** Vite + `vite-plugin-logseq` is common; Parcel also viable. Manual Rollup config is complex.
        *   **Initialization:** Can include `await`, but needs build tool support; avoiding top-level `await` or delaying execution is safer.
        *   **Error Handling:** All use `.catch(console.error)`.
    *   Updated '[Plugin Writing Guide.md](http://xn--5nq07djxb7y5a72as40i.md/)'.
5.  Switched back to Vite + `vite-plugin-logseq` (Based on researched best practices):
    *   Uninstalled Rollup, installed Vite (`3.2.2`) and `vite-plugin-logseq` (`1.1.2`).
    *   Reset `package.json` (omitted `logseq.main`, defined root `main`, Vite build script).
    *   Created `vite.config.ts` (later changed to `.cjs`) using `logseqDevPlugin()`.
    *   Created `index.html`.
    *   Rewrote `index.ts` to implement full functionality.
    *   **Issue:** Encountered persistent build error `TypeError: logseqDevPlugin is not a function`.
6.  Attempted to fix Vite build error (Environment investigation):
    *   Tried different import syntaxes, config file formats (`.ts` vs `.cjs`), `package.json` `type` field.
    *   Cleared npm cache, deleted `node_modules`/`package-lock.json` and reinstalled.
    *   Switched Node.js version to LTS (v20.19.0).
    *   **Result:** All attempts **failed**; the Vite build error persisted. Conclusion: The problem likely originates from the local development environment.
7.  Switched back again to Parcel (v2.9.3 - referencing `pomodoro-plugin`):
    *   Uninstalled Vite, installed Parcel (`2.9.3`).
    *   Restored Parcel's `package.json` (no `main`, has `targets`).
    *   Restored Parcel's `index.html` (with `type="module"`).
    *   Used the full-featured `index.ts` (adjusted initialization pattern, avoided top-level `await`).
    *   Cleared Parcel cache (`.parcel-cache`).
    *   **Result:** Parcel build **succeeded**.
8.  Final Test (Parcel v2.9.3 + Full/Simplified/Pure JS code):
    *   Used standard loading method (Method C).
    *   **Observation:** Plugin was recognized (icon displayed), but **JavaScript code did not execute at all** (no functionality, no console logs), even when simplified to just `console.log("Hello")` or `logseq.ready(() => console.log('Ready'))`.
    *   **Comparison:** `logseq-pomodoro-plugin` worked normally in the same environment and loading method.

------

## Final Conclusion

After exhaustive investigation, the root cause is highly likely a **subtle compatibility issue or Bug between Logseq 0.10.9 and the JavaScript output generated by Parcel v2.9.3 for this specific project (`logseq-word-count-plugin`)**. This prevents Logseq from properly initializing and executing the plugin's JS code in this environment (the `logseq.ready` callback is not triggered). Since other similar plugins function correctly and all reasonable code and configuration fixes have been attempted, this problem appears to be beyond the scope of conventional debugging.
